File: backend/src/main/java/com/yoursunshine/backend/service/GroupService.java
package com.yoursunshine.backend.service;

import com.yoursunshine.backend.endpoint.dto.GroupCreateDto;
import com.yoursunshine.backend.endpoint.dto.GroupDetailDto;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface GroupService {

    /**
     * Create a new group
     * @param room_id the id of the room to create the group in
     * @param group the group to create
     * @return the created group
     */
    GroupDetailDto create(Long room_id, GroupCreateDto group);

    /**
     * Get all groups in a room
     * @param room_id the id of the room to get the groups from
     * @return all groups
     */
    List<GroupDetailDto> getAll(Long room_id);

    /**
     * Update a group
     * @param id the id of the group to update
     * @param group the group to update
     * @return the updated group
     */
    GroupDetailDto update(Long id, GroupDetailDto group);

    /**
     * Delete a group
     * @param id the id of the group to delete
     */
    void delete(Long id);
}

File: frontend/src/app/services/group.service.ts
import { Injectable } from '@angular/core';
import {Globals} from "../global/globals";
import {HttpClient} from "@angular/common/http";
import {Group} from "../dtos/group";
import {Observable} from "rxjs";

@Injectable({
  providedIn: 'root'
})
export class GroupService {

  private groupBaseUri: string = this.globals.backendUri + '/rooms';

  constructor(
    private http: HttpClient,
    private globals: Globals,
  ) { }

  /**
   * Creates new group
   * @param room_id the room id
   * @param group to create
   * @return the created group
   */
  create(room_id: number, group: Group): Observable<Group>{
    console.log('create group: ', group);
    return this.http.post<Group>(this.groupBaseUri + '/' + room_id + "/groups", group);
  }

  /**
   * Returns all groups
   * @param room_id the room id
   * @return all groups
   */
  getAll(room_id: number): Observable<Group[]> {
    console.log('get all groups');
    return this.http.get<Group[]>(this.groupBaseUri + '/' + room_id + "/groups");
  }

  /**
   * Updates the group
   * @param room_id the room id
   * @param group the group to update
   * @return the updated group
   */
  update(room_id: number, group: Group): Observable<Group> {
    console.log('update group: ', group);
    return this.http.put<Group>(this.groupBaseUri + '/' + room_id + "/groups" + '/' + group.id, group);
  }

  /**
   * Deletes the group
   * @param room_id the room id
   * @param id of the group to delete
   */
  delete(room_id: number, id: number): Observable<any> {
    console.log('delete group: ', id);
    return this.http.delete<any>(this.groupBaseUri + '/' + room_id + "/groups" + '/' + id);
  }
}
